src/ArrayUtility.php

```diff
@@ -119,13 +119,72 @@ public static function dotRead(array $array = null, $key, $default_value = null)
 		}
 
 		//Get a default value
-		if (!isset($value) && isset($default_value)) {
+		if ($value === null && $default_value !== null) {
 			return $default_value;
 		}
 
 		return $value;
 	}
 
+    /**
+     * Write a value to a nested array
+     * @param array  $array
+     * @param string $writeKey The write key e.g. "level1.level2.key"
+     * @param mixed  $value    Value to write
+     * @return bool  Returns true if write was successful
+     */
+    public static function dotWrite(array &$array, $writeKey, $value = null)
```

Comment:
Should be immuteable - how come array ref is getting passed in?

Reply:
Should it be immutable? Felt like I wanted to mutate the array without causing a copy op.

Reply:
All the others are immutable

Reply:
Added ::dotMutate, and changed ::dotWrite